src/appleseed.shared/application/progresstilecallback.cpp

```diff
@@ -80,12 +80,19 @@ namespace
 
         void on_tiled_frame_begin(const Frame* frame) override
         {
-            m_stopwatch.start();
+            // Do not restart the stopwatch if it is already running.
+            if (m_rendered_pixels == 0.0)
+                m_stopwatch.start();
         }
 
         void on_tiled_frame_end(const Frame* frame) override
         {
-            m_stopwatch.clear();
+            const size_t total_tiles = frame->image().properties().m_tile_count * m_pass_count;
+
+            // Clear the stopwatch only when the number of rendered tiles becomes equal to total tiles.
+            // This makes sure that we don't clear the stopwatch in between passes.
+            if (m_rendered_tiles == total_tiles)
+                m_stopwatch.clear();
```

Comment:
Why do we even need to clear the stopwatch?

Reply:
To free up resources?
There is no reason other than that. Would be useful if want to show render time estimation for post-processing stages, but not required right now.

Reply:
If you go look at it, `Stopwatch::clear()` doesn't free up resources, it simply resets the elapsed time to zero. I suggest removing the whole `on_tiled_frame_end()` method.